packages/playwright-core/src/server/recorder/codeGenerator.ts

```diff
@@ -130,7 +130,9 @@ export class CodeGenerator extends EventEmitter {
       this._currentAction.action.signals.push(signal);
       return;
     }
-    if (this._lastAction && !this._lastAction.committed) {
+
+    // Consider fill action as committed for the purposes of navigation signal processing.
+    if (this._lastAction && !this._lastAction.committed && this._lastAction.action.name !== 'fill') {
```

Comment:
It also happens in a click followed by a navigation.
In my test, if you reorder the calls, you'll have the same issue:

```ts
  await page.goto(server.PREFIX + '/input/button.html');
  await page.click('button');
  await page.goto(server.PREFIX + '/input/textarea.html');
  await page.fill('textarea', 'hello world');
```

I understand that the navigation can be interpreted as the effect of clicking the button if they're almost at the same time, so I'm not sure if there's a way to distinguish a navigation caused by a click from a navigation caused by entering a URL in the navigation bar.

Reply:
You are right, I thought that we commit clicks sooner than other actions. I don't want to reduce the commit timer and I am not sure I want to commit on mouse move, so I guess existing imperfection is the lesser evil. I'll close the issue and the PR.